lib/ronin/app/validations/vulns_params.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+#
+# ronin-app - a local web app for Ronin.
+#
+# Copyright (C) 2023 Hal Brodigan (postmodern.mod3@gmail.com)
+#
+# ronin-app is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ronin-app is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with ronin-app.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+require 'dry/validation'
+require 'ronin/app/types/spider'
+
+module Ronin
+  module App
+    module Validations
+      #
+      # Validations for form params submitted to `POST /vulns`.
+      #
+      class VulnsParams < Dry::Validation::Contract
+
+        params do
+          required(:url).filled(:string)
+
+          optional(:lfi).hash do
+            optional(:os).maybe(:string)
+            optional(:depth).maybe(:integer)
+            optional(:filter_bypass).maybe(:string)
+          end
+
+          optional(:rfi).hash do
+            optional(:filter_bypass).maybe(:string)
+            optional(:test_script_url).maybe(:string)
+          end
+
+          optional(:sqli).hash do
+            optional(:escape_quote).maybe(:bool)
+            optional(:escape_parens).maybe(:bool)
+            optional(:terminate).maybe(:bool)
+          end
+
+          optional(:ssti).hash do
+            optional(:escape).maybe(:string) # ?
```

Comment:
We should probably use an enum type here. To define an enum type that can be used by both `Validations::VulnsParams` and `Workers::Vulns::Params`, it should be defined in `Ronin::App::Types::Vulns`/`lib/ronin/app/types/vulns.rb` (see the other files within `lib/ronin/app/types/` for examples).

```ruby
require 'ronin/app/types'

module Ronin
  module App
    module Types
      #
      # Types for {Validations::VulnsParams}.
      #
      module Vulns
        module LFI
          # The type of OS
          OSType = Types::String.enum('unix', 'windows')
          ...
        end
        
        ...
        
        module SSTI
          # The type of SSTI escape expression
          EscapeType = Types::String.enum('double_curly_braces', ...)
        end
      end
    end
  end
end
```

Then use `Types::Vulns::LFI::OSType` instead of `:string` in `Validations::VulnsParams` and `Workers::Vulns::Params`.